src/python/pants/backend/jvm/tasks/jvm_compile/rsc/rsc_compile.py

```diff
@@ -0,0 +1,586 @@
+          #     do multiple cache entries per target-task tuple.
+          on_success=ivts.update,
+          on_failure=ivts.force_invalidate,
+        )
+      )
+
+    return rsc_jobs + zinc_jobs
+
+  def select_runtime_context(self, ccs):
+    return ccs[1]
+
+  def create_compile_context(self, target, target_workdir):
+    # workdir layout:
+    # rsc/
+    #   - index/   -- metacp results
+    #   - outline/ -- semanticdbs for the current target as created by rsc
+    #   - m.jar    -- reified scala signature jar
+    # zinc/
+    #   - classes/   -- class files
+    #   - z.analysis -- zinc analysis for the target
+    #   - z.jar      -- final jar for the target
+    #   - zinc_args  -- file containing the used zinc args
+    sources = self._compute_sources_for_target(target)
+    rsc_dir = os.path.join(target_workdir, "rsc")
+    zinc_dir = os.path.join(target_workdir, "zinc")
+    return [
+      RscCompileContext(
+        target=target,
+        analysis_file=None,
+        classes_dir=None,
+        jar_file=None,
+        zinc_args_file=None,
+        rsc_mjar_file=os.path.join(rsc_dir, 'm.jar'),
+        log_dir=os.path.join(rsc_dir, 'logs'),
+        sources=sources,
+        index_dir=os.path.join(rsc_dir, 'index'),
+        outline_dir=os.path.join(rsc_dir, 'outline'),
+      ),
+      CompileContext(
+        target=target,
+        analysis_file=os.path.join(zinc_dir, 'z.analysis'),
+        classes_dir=os.path.join(zinc_dir, 'classes'),
+        jar_file=os.path.join(zinc_dir, 'z.jar'),
+        log_dir=os.path.join(zinc_dir, 'logs'),
+        zinc_args_file=os.path.join(zinc_dir, 'zinc_args'),
+        sources=sources,
+      )
+    ]
+
+  def _runtool(self, main, tool_name, args, distribution, tgt=None, input_files=tuple(), output_dir=None):
+    if self.execution_strategy == self.HERMETIC:
+      # TODO: accept input_digests as well as files.
+      with self.context.new_workunit(tool_name) as wu:
+        tool_classpath_abs = self.tool_classpath(tool_name)
+        tool_classpath = fast_relpath_collection(tool_classpath_abs)
+
+        pathglobs = list(tool_classpath)
+        pathglobs.extend(input_files)
+        root = PathGlobsAndRoot(
+          PathGlobs(tuple(pathglobs)),
+          text_type(get_buildroot()))
+
+        tool_snapshots = self.context._scheduler.capture_snapshots((root,))
+        input_files_directory_digest = tool_snapshots[0].directory_digest
+        classpath_for_cmd = os.pathsep.join(tool_classpath)
+        cmd = [
+          distribution.java,
+        ]
+        cmd.extend(self.get_options().jvm_options)
+        cmd.extend(['-cp', classpath_for_cmd])
+        cmd.extend([main])
+        cmd.extend(args)
+
+        epr = ExecuteProcessRequest(
+          argv=tuple(cmd),
+          env=dict(),
+          input_files=input_files_directory_digest,
+          output_files=tuple(),
+          output_directories=(output_dir,),
+          timeout_seconds=15*60,
```

Comment:
Is there something special about a timeout of 15 minutes, or it's picked simply as a period of time that is reasonably long?

Reply:
It's the reasonably long thing. It may be a current default somewhere. At some point, we should probably thread config through for it, but this patch doesn't do that.